File: MainApp/ios/MiniAppViewManager.m
#import <React/RCTViewManager.h>

@interface RCT_EXTERN_REMAP_MODULE(MiniApp, MiniAppViewManager, RCTViewManager)
RCT_EXPORT_VIEW_PROPERTY(source, NSDictionary)
RCT_EXPORT_VIEW_PROPERTY(onStartLoad, RCTDirectEventBlock)
RCT_EXPORT_VIEW_PROPERTY(onEndLoad, RCTDirectEventBlock)
RCT_EXPORT_VIEW_PROPERTY(onError, RCTDirectEventBlock)

RCT_EXTERN_METHOD(dispatch:(nonnull NSNumber *)reactTag)
@end

File: MainApp/ios/MiniAppViewManager.swift
import React

@objc(MiniAppViewManager)
class MiniAppViewManager: RCTViewManager {
  override static func requiresMainQueueSetup() -> Bool {
    return true
  }
  
  override func view() -> UIView! {
    return MiniAppView()
  }
  
  @objc(dispatch:)
  func dispatch(reactTag: NSNumber) {
    bridge.uiManager.prependUIBlock({_ , viewRegistry in
      let view = viewRegistry?[reactTag]
      if !(view is MiniAppView) {
        print("Invalid view returned from registry, expecting RCTVideo, got: ", String(describing: view))
      } else if let view = view as? MiniAppView {
        view.dispatch()
      }
    })
  }
}
